config.toml.example

```diff
@@ -329,6 +329,12 @@
 # Defaults to rust.debug-assertions value
 #debug-assertions-std = false
 
+# Whether or not to leave debug! and trace! calls in the rust binary.
+# Overrides the `debug-assertions` option, if defined.
+# 
+# Defaults to rust.debug-assertions value
```

Comment:
Hm, so this brings up two interesting questions:
 * Should we default to this being on? If you're doing a lot of rustc development, it can certainly be helpful to have debug logs, but it can also be a pretty annoying slowdown (IIRC, benchmarks show 5-10%).
 * Show we tie to to the old way of turning it on, i.e., debug-assertions?

cc @jyn514, we might want an MCP or similar here to figure out what people think good defaults would be.

Reply:
I think it should start as off-by-default. There was a lot of discussion in https://github.com/rust-lang/rust/pull/76588 about enabling logging and it was almost a 10% perf hit on some benchmarks: https://perf.rust-lang.org/compare.html?start=22e6099330cde0e7b1529774fe27874f8326de7a&end=ededceef5651050bda1a2058af270da498bb1333.

That said, I would _love_ to get the perf impact low enough that this can be off by default (or even always on).

Reply:
> That said, I would love to get the perf impact low enough that this can be on by default (or even always on).

One possible way to do this is to separate `debug!` from `trace!` and have _that_ be opt-in instead. AFAIK most of the compiler uses debug! currently, but I separate the two in rustdoc and it's worked very well. In particular, `collect_intra_doc_links` prints ludicrous amounts of info with `trace` but only ~15 lines or so per module with `debug`: https://github.com/rust-lang/rust/blob/master/src/librustdoc/passes/collect_intra_doc_links.rs#L647

Reply:
As it is implemented now, it should be off by default, and on wit debug-assertions, which matched exactly the semantics of the current config.toml